### PR TITLE
Add support for TemporalMemoryCPP in TMRegion

### PR DESCRIPTION
**Pull request by subutai**
Fix bug in getConstructorArguments in temporal_memory_factory
Use the proper accessor methods to get active/predictive cells
Validate with hello_classification_model

### CODE REVIEW

**Comment by subutai** [comment]
@oxtopus Can you validate that I did the right thing in getConstructorArguments? I believe the previous version didn't do the right thing (it was a copy of createModel)

**Comment by oxtopus** [comment]
Hmm.  yes, the status quo `getConstructorArguments()` looks suspiciously broken.  Your fix looks good, assuming you're _only_ concerned with args and not the full function signature.  Since `inspect.getargspec()` returns a named tuple, you may want to access it by name instead of index:

```
return inspect.getargspec(getattr(TemporalMemoryTypes, modelName).__init__).args
```

**Comment by subutai** [comment]
@oxtopus Thanks. The comments string said it was to return a list of argument names only, but I think additional information would be useful. I don't think anyone else is using it, so I can make the change easily. Does this look ok?

**Comment by oxtopus** [comment]
Something else to consider, since `__init__()` is an instance method, its argspec includes `self`.  So you may need to strip that out.

**Comment by subutai** [comment]
@oxtopus Yeah, I had put that in the comments, but I think it is cleaner to strip it out so just pushed that. Thanks.

File: htmresearch/algorithms/temporal_memory_factory.py [hunk 1]
```diff
@@ -77,14 +77,25 @@ def createModel(modelName, **kwargs):
...
+                                 item is usually 'self'.
+  @return defaults (list)        a list of default values for each argument
+                                 (excluding self)
   """
 
   if modelName not in TemporalMemoryTypes.getTypes():
     raise RuntimeError("Unknown model type: " + modelName)
 
-  return inspect.getargspec(getattr(TemporalMemoryTypes, modelName).__init__)[0]
+  return (
```
Comment (oxtopus) [inline htmresearch/algorithms/temporal_memory_factory.py]: Better to not repeat the call to `inspect.getargspec()`.  Instead, do something like:

```python
argspec = inspect.getargspec(...)
return (argspec.args, argspec.defaults)
```

or:

```python
from operator import attrgetter
...
return attrgetter("args", "defaults")(inspect.getargspec(...))
```

**Comment by oxtopus** [comment]
👍  pending deduping.

**Pull request merged**